Automattic/wp-calypso
# Gutenberg: Enable Jetpack Gutenframe in the desktop app

#### Changes proposed in this Pull Request

This PR enables Gutenframe for Jetpack/Atomic sites in the desktop app by removing the `jetpack/gutenframe` feature toggle (since all environments will have it enabled now).

In order to keep the user in the app, it also fallbacks to the classic editor when the site is not eligible for Gutenframe (i.e. because a denied plugin is active) rather than opening WP Admin in the browser. This was previously controlled by the `calypsoify/gutenberg` feature flag which has been renamed to `gutenberg/redirect` in order to align the name with the value of the selected editor.

#### Testing instructions

This needs to be tested on the desktop app, so make sure you have the dev environment [installed and running](https://github.com/Automattic/wp-desktop/blob/468ecd4d68cdafbc2935facfcd15953f46db98ce/docs/install.md) (more info about how to set up the secrets at paaHJt-7y-p2).

- In the desktop repo, point the calypso submodule to this branch: `cd calypso && git fetch && git checkout update/32048-gutenframe-enable-desktop && cd ..`.
- Generate  a new build: `make build`.
- Open the built app in the `release/<YOUR_OS>` folder (i.e. `release/mac/WordPress.com.app`).
- Switch to an Atomic site.
- Start creating a post.
- Make sure the block editor is loaded in the desktop app.
- Install a denied plugin (see PCYsg-lOu-p2 for information about how to get the current list of denied plugins).
- Close the desktop app and open it again, so the cache is cleared and the new installed plugin is considered when determining which editor should be loaded.
- Create a new post again.
- Verify the Calypso classic editor is loaded now in the desktop app:
<img width="1415" alt="Screen Shot 2019-07-24 at 08 16 27" src="https://user-images.githubusercontent.com/1233880/61753736-5a92ca00-adeb-11e9-87b4-f840eb0603b5.png">

Fixes #32048
